dev/error_message_refiner.py

```diff
@@ -0,0 +1,265 @@
+    file = random.choice(files)
+    return file
+
+
+def _find_function(file_name: str, search_string: str) -> Optional[str]:
+    """
+    Searches for a function in the given file containing the specified search string.
+    Returns the name of the function if found, otherwise None.
+    """
+    with open(file_name, "r") as file:
+        content = file.read()
+        functions = re.findall(r"def\s+(\w+)\s*\(", content)
+
+        for function in functions:
+            function_content = re.search(
+                rf"def\s+{re.escape(function)}(?:(?!def).)*?{re.escape(search_string)}",
+                content,
+                re.DOTALL,
+            )
+            if function_content and search_string in function_content.group(0):
+                return function
+
+    return None
+
+
+def _find_func_body(file_name: str, search_string: str) -> Optional[str]:
+    """
+    Searches for a function body in the given file containing the specified search string.
+    Returns the function body if found, otherwise None.
+    """
+    with open(file_name, "r") as file:
+        content = file.read()
+        functions = re.findall(r"def\s+(\w+)\s*\(", content)
+
+        for function in functions:
+            function_content = re.search(
+                rf"def\s+{re.escape(function)}(?:(?!def\s).)*?{re.escape(search_string)}",
+                content,
+                re.DOTALL,
+            )
+            if function_content and search_string in function_content.group(0):
+                return function_content.group(0)
+
+    return None
+
+
+def _get_error_function(error_class: str) -> str:
+    """
+    Retrieves the name of the error function that triggers the given error class.
+    """
+    search_string = error_class
+    matched_file = _git_grep_files(search_string)
+    err_func = _find_function(matched_file, search_string)
+    return err_func
+
+
+def _get_source_code(error_class: str) -> str:
+    """
+    Retrieves the source code of a function where the given error class is being invoked.
+    """
+    search_string = error_class
+    matched_file = _git_grep_files(search_string)
+    err_func = _find_function(matched_file, search_string)
+    source_file = _git_grep_files(err_func, exclude=matched_file)
+    source_code = _find_func_body(source_file, err_func)
+    return source_code
+
+
+def _get_error_message(error_class: str) -> str:
+    """
+    Returns the error message from the provided error class
+    listed in core/src/main/resources/error/error-classes.json.
+    """
+    with open(PATH_TO_ERROR_CLASS) as f:
+        error_classes = json.load(f)
+
+    error_message = error_classes.get(error_class)
+    if error_message:
+        return error_message["message"]
+    else:
+        return f"Error message not found for class: {error_class}"
+
+
+def ask_chat_gpt(error_class: str, model_name: str) -> Tuple[str, str]:
+    """
+    Requests error message improvement from Chat GPT.
```

Comment:
> Requests error message improvement from Chat GPT.

```suggestion
    Requests error message improvement from GPT API
```

Reply:
^ change all "chat gpt" here to gpt API? to be more accurate

Reply:
Applied. Thanks!